spec/integration/admin/grove_monitor_spec.rb

```diff
@@ -2,7 +2,21 @@
 include Pages::Authentication
 
 RSpec.feature 'Grove Monitor' do
-  let(:teacher) { create(:teacher) }
+  let(:grove) { create(:grove_with_students) }
+  let(:teacher) { grove.teachers.first }
+  let(:student1) { grove.students.first }
+  let(:student2) { grove.students.last }
+  let(:location) { create(:location) }
+  let(:location2) { create(:location) }
+  let(:grove2) { create(:grove_with_resources) }
+  let(:teacher2) { grove2.teachers.first }
+  let(:student3) { grove2.students.first }
+  let(:student4) { grove2.students.last }
+  let(:event) { student3.events.first }
+  let(:event2) { student4.events.first }
+  let(:scan) { event.scans.first }
+  let(:scan2) { event2.scans.last }
+
```

Comment:
when you have this much setup, it might be telling you that you're testing the intricacies of this at the wrong level. I don't see any reason to change this, but I think this indicates that we should be breaking the problem down into smaller chunks and testing those at the unit level. We're pretty light on that side of things in the PR.